packages/headless/src/features/case-assist/case-assist-analytics-actions.ts

```diff
@@ -31,128 +36,249 @@ export const logCaseNextStage = (
+    __legacy__getBuilder: (client, state) => {
+      return client.logCaseCancelled({
         ticket: caseAssistCaseSelector(state),
-      })
-  );
+      });
+    },
+    analyticsType: 'CaseAssist.Cancel',
+    analyticsPayloadBuilder: (): CaseAssist.Cancel => ({
+      reason: 'quit',
+    }),
+  });
 
 export const logUpdateCaseField = (fieldName: string): CaseAssistAction =>
-  makeCaseAssistAnalyticsAction(
-    'analytics/caseAssist/case/field/update',
-    (client, state) =>
-      client.logUpdateCaseField({
+  makeCaseAssistAnalyticsAction({
+    prefix: 'analytics/caseAssist/case/field/update',
+    __legacy__getBuilder: (client, state) => {
+      return client.logUpdateCaseField({
         fieldName,
         ticket: caseAssistCaseSelector(state),
-      })
-  );
+      });
+    },
+    analyticsType: 'CaseAssist.UpdateField',
+    analyticsPayloadBuilder: (state): CaseAssist.UpdateField => {
+      const fieldValue =
+        state.caseField?.fields?.[fieldName]?.value ||
+        state.caseInput?.[fieldName]?.value ||
+        '';
+      return {
+        fieldName,
+        fieldValue,
+      };
+    },
+  });
 
 export const logAutoSelectCaseField = (
   classificationId: string
 ): CaseAssistAction =>
-  makeCaseAssistAnalyticsAction(
-    'analytics/caseAssist/classification/click',
-    (client, state) =>
-      client.logSelectFieldSuggestion({
+  makeCaseAssistAnalyticsAction({
+    prefix: 'analytics/caseAssist/classification/click',
+    __legacy__getBuilder: (client, state) => {
+      return client.logSelectFieldSuggestion({
         suggestion: caseAssistCaseClassificationSelector(
           state,
           classificationId,
           true
         ),
         ticket: caseAssistCaseSelector(state),
-      })
-  );
+      });
+    },
+    analyticsType: 'CaseAssist.SelectFieldClassification',
+    analyticsPayloadBuilder: (state): CaseAssist.SelectFieldClassification => {
+      return {
+        autoselected: true,
+        fieldClassification: {
+          id: classificationId,
+          responseId: state.caseField?.status?.lastResponseId || '',
+        },
+      };
+    },
+  });
 
 export const logClassificationClick = (
   classificationId: string
 ): CaseAssistAction =>
-  makeCaseAssistAnalyticsAction(
-    'analytics/caseAssist/classification/click',
-    (client, state) =>
-      client.logSelectFieldSuggestion({
+  makeCaseAssistAnalyticsAction({
+    prefix: 'analytics/caseAssist/classification/click',
+    __legacy__getBuilder: (client, state) => {
+      return client.logSelectFieldSuggestion({
         suggestion: caseAssistCaseClassificationSelector(
           state,
           classificationId
         ),
         ticket: caseAssistCaseSelector(state),
-      })
-  );
+      });
+    },
+    analyticsType: 'CaseAssist.SelectFieldClassification',
+    analyticsPayloadBuilder: (state): CaseAssist.SelectFieldClassification => {
+      return {
+        autoselected: false,
```

Comment:
Do we just omit it? Or do we have to be explicit that it's "not" autoselected?

Reply:
yes we need to be explicit.